contrib/babelfishpg_tsql/src/collation.c

```diff
@@ -482,7 +514,7 @@ transform_from_ci_as_for_likenode(Node *node, OpExpr *op, like_ilike_info_t like
 											oprfuncid(optup));
 		ReleaseSysCache(optup);
 		/* construct pattern||E'\uFFFF' */
-		highest_sort_key = makeConst(TEXTOID, -1, InvalidOid, -1,
```

Comment:
Line 505/506 should be converted to error condition. It is always expected that we will find all the required operators - 
```
if (optup == (Operator) NULL)
			return node;
```

Reply:
same for 522/523 line
```		
if (optup == (Operator) NULL)
			return node;

```

Reply:
That should not be the case, if we can not find any compatible operator then we should be using LIKE directly (which can be then handled by PG engine if there is any issue). As this is an optimisation and NOT measure of correctness, putting such restriction is NOT ideal .